### PR TITLE
Updates for SDK v5.1

### PR DESCRIPTION
**Pull request by Mriegel**

### CODE REVIEW

**Comment by kbataran** [comment]
can you update the README to include cordova installation instructions/developer setup? Similar to https://github.com/localytics/react-native-template-app

**Comment by kbataran** [comment]
Probably as simple as saying `$ sudo npm install -g cordova` and linking to https://cordova.apache.org/docs/en/latest/guide/cli/

File: platforms/android/res/values/localytics.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <string name="ll_default_push_channel_name" translatable="false">Default</string>
+    <!--Configure the default channel description used for push notifications-->
+    <string name="ll_default_push_channel_description" translatable="false"/>
+
+    <!--Configure the default channel used for places notifications-->
+    <string name="ll_default_places_channel_id" translatable="false">localytics_default</string>
+    <!--Configure the default channel name used for places notifications-->
+    <string name="ll_default_places_channel_name" translatable="false">Default</string>
+    <!--Configure the default channel description used for places notifications-->
+    <string name="ll_default_places_channel_description" translatable="false"/>
```
Comment (kbataran) [inline platforms/android/res/values/localytics.xml:42]: This is missing the location update settings (prob need to grab the latest localytics.xml)

File: plugins/com.localytics.phonegap.LocalyticsPlugin/README.md [hunk 2]
```diff
@@ -444,5 +444,6 @@ Localytics.setPushMessageConfiguration(pushConfig);
 Localytics.setPlacesMessageConfiguration(placesConfig);
 ```
 
-## SampleApp
-The Cordova app found under SampleApp demonstrates a list of functional APIs that can be called via the JavaScript interface. Update "LocalyticsAppKey" and "LOCALYTICS_APP_KEY" for their respective platforms to easily test out the calls.
+## Sample App
+
+A sample template app is available at [https://github.com/localytics/cordova-template-app](https://github.com/localytics/cordova-template-app).
```
Comment (kbataran) [inline plugins/com.localytics.phonegap.LocalyticsPlugin/README.md:459]: another good example of why this should prob be ignored

**Review by kbataran — COMMENTED** [review]
1) Should we just integrate FCM as the standard in these template apps?
2) Should the "plugins" folder be .gitignored similar to how the "node_modules" folder is for React Native apps?

File: platforms/android/res/values/localytics.xml (hunk 1 above)
Comment (Mriegel) [reply platforms/android/res/values/localytics.xml:42]: They aren't required, so I wouldn't worry about it.

File: platforms/android/platform_www/plugins/com.localytics.phonegap.LocalyticsPlugin/www/localytics.js [hunk 3]
```diff
@@ -454,19 +481,32 @@ Localytics.prototype.setInAppMessageConfiguration = function (config) {
 	cordova.exec(null, null, "LocalyticsPlugin", "setInAppMessageConfiguration", [config]);
 }
 
-// iOS only: Returns whether the ADID parameter is added to In-App call to action URLs
```
Comment (abashyam) [inline platforms/android/platform_www/plugins/com.localytics.phonegap.LocalyticsPlugin/www/localytics.js:457]: the Cordova api should be same for android and iOS.

File: platforms/ios/LocalyticsSample/Classes/AppDelegate.m [hunk 4]
```diff
@@ -27,13 +27,38 @@ Licensed to the Apache Software Foundation (ASF) under one
...
 #import "MainViewController.h"
+@import UserNotifications;
+@import Localytics;
 
 @implementation AppDelegate
 
 - (BOOL)application:(UIApplication*)application didFinishLaunchingWithOptions:(NSDictionary*)launchOptions
 {
     self.viewController = [[MainViewController alloc] init];
+    if (NSClassFromString(@"UNUserNotificationCenter")) {
```
Comment (abashyam) [inline platforms/ios/LocalyticsSample/Classes/AppDelegate.m:38]: Use weak linking which ensures the same behavior.This won't detect typos.

File: platforms/ios/Podfile.lock [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
 
 SPEC CHECKSUMS:
-  LocalyticsStaticFramework: 6eaa42ea2d6e8f331831afd8e9656c3cbc238f2e
+  Localytics: 420a187fd1ba94d4a2afff66e100d1569389d9e4
 
-PODFILE CHECKSUM: ef8a32be96bd35bc32ddbd1d80542ef3258ce9ae
+PODFILE CHECKSUM: b65dd1216182a232999d9c79dbd9a33691d06525
 
-COCOAPODS: 1.3.1
+COCOAPODS: 1.5.0
```
Comment (abashyam) [inline platforms/ios/Podfile.lock:16]: Should be 1.5.1

File: platforms/ios/www/index.html [hunk 6]
```diff
@@ -92,274 +92,295 @@ <h1>Localytics Debug</h1>
...
+                        <hr class="hr_break">
+                        inc/dec: <input id="inputProfileAttributeIncrementValue" type="number" min="1" max="50" value="1">
+                            <button id="btnIncrementProfile" class="localytics_button increment_profile">Increment</button>
+                            <button id="btnDecrementProfile" class="localytics_button decrement_profile">Decrement</button>
+                            <hr class="hr_break">
+                            <br><br>
+                            
+                            
+                            <div class="label">User Info (Set)</div>
+                            key: <input id="inputIdentifierKey" type="text" value="key1" style="width:40px">
```
Comment (abashyam) [inline platforms/ios/www/index.html:171]: Something wrong with these indents.

File: www/js/index.js [hunk 7]
```diff
@@ -44,15 +44,15 @@ var app = {
     onIntegrate: function () {
         document.addEventListener("resume", app.onResume, false);
         document.addEventListener("pause", app.onPause, false);
-        Localytics.integrate("YOUR-LOCALYTICS-APP-KEY");
+        Localytics.integrate("dcdf9cb70fff87655848357-8f7e9180-385a-11e4-a3d0-009c5fda0a25");
```
Comment (abashyam) [inline www/js/index.js]: Dont like the appKey here. the Old code is right.

**Review by abashyam — CHANGES_REQUESTED** [review]
I will test this on a device once, I get into office and approve it. appKey is the only item of concern.

File: www/js/index.js (hunk 7 above)
Comment (Mriegel) [reply www/js/index.js]: ya just a mistake after testing - will fix

File: platforms/android/platform_www/plugins/com.localytics.phonegap.LocalyticsPlugin/www/localytics.js (hunk 3 above)
Comment (Mriegel) [reply platforms/android/platform_www/plugins/com.localytics.phonegap.LocalyticsPlugin/www/localytics.js:457]: Not sure I follow... it is the same for Android and iOS (it used to only exist on iOS)

**Comment by kbataran** [comment]
I still think there's some work that could be done with the README... making the instructions for integration clearer to developers who are using this repo to build their own cordova apps. The README in the React Native template app might be a good place for inspiration, let me know if you think the README is fine as it is

**Comment by Mriegel** [comment]
I just mentally diff-ed the react-native README and this one.  Not sure I see what you feel is missing.

**Review by kbataran — COMMENTED** [review]
Two things to be addressed before I'll approve:
1) _autoIntegrate_ button/function does not work
2) You need to remove and link the LocalyticsPlugin twice in order to run the sample app

**Comment by Mriegel** [comment]
Fixed the autoIntegrate option on iOS. Should be able to fix the other issue once we release 5.1.  Will hold off on merging until then.

**Pull request merged**